src/Microsoft.FeatureManagement/FeatureManager.cs

```diff
@@ -187,6 +218,52 @@ await featureFilter.EvaluateAsync(context).ConfigureAwait(false) == targetEvalua
             return enabled;
         }
 
+        private void BindSettings(IFeatureFilterMetadata filter, FeatureFilterEvaluationContext context)
+        {
+            IFilterParametersBinder binder = filter as IFilterParametersBinder;
+
+            if (binder == null)
```

Comment:
What do you think about having a separate short circuit like

```
if (!(_featureDefinitionProvider is IFeatureDefinitionProviderCacheable))
{
    context.Settings = binder.BindParameters(context.Parameters);

    return;
}
```

Suggesting it because any interaction with cache key or cache doesn't make sense if the definition provider isn't cachable.

Reply:
Good idea, makes the cache if statement a little cleaner as a result.